libs/perun/components/src/lib/users-dynamic-list/users-dynamic-list.component.ts

```diff
@@ -100,11 +109,15 @@ export class UsersDynamicListComponent implements OnInit, OnChanges, AfterViewIn
     if (!this.authResolver.isPerunAdminOrObserver()) {
       this.displayedColumns = this.displayedColumns.filter((column) => column !== 'id');
     }
+    if (this.includeConsents) {
+      this.displayedColumns.push('consentStatus');
```

Comment:
You should add the column `consentStatus` also to the function `getExportDataForColumn` for the correct export of data. I am not sure if it is the best solution, but you can make the function `getExportDataForColumn` non-static and an arrow function:
```
getExportDataForColumn = (data: RichUser, column: string): string => {
```
If you change it in this way, you can then use this reference in the function and you can add this case to switch (you need to add your pipe to the constructor and providers):
```
case 'consentStatus':
        return this.consentStatusPipe.transform(data, this.consents);
```
After that, you have to change `UsersDynamicListComponent.getExportDataForColumn` reference to `this.getExportDataForColumn` in this file.

As I wrote I am not sure if it is a 100 % correct solution, but IMHO it works. If you find a better solution, I will be happy.

Reply:
Other solution that comes to mind is to create a new object that will hold a User with his corresponding consent status and then get the status for each user in the export methods before calling getDataForExport. Would be a bit more code, but getDataForColumn would remain static (which is imperative i guess?)

Reply:
Yes, the new object should resolve this use case. You just have to be careful during the refactoring since two components currently use this list. It shouldn't be a problem, just saying to don't forget to test both of them.